[nmstate-1.4] dns: Fix DNS option `ndots`, `timeout` and `attempts`

cathay4t opened the pull request:
## Patch  1
The `ndots`, `timeout` and `attempts` DNS options are allowed to
hold a integer value in the format of `<opt_name>:<int>`. Previously,
nmstate is treating them as invalid DNS option. This patch fix it.

Now this YAML is supported:

```yml
dns-resolver:
  config:
    options:
    - rotate
    - ndots:9
```



## Patch 2

When user desires:

```yml
---
dns-resolver:
  config:
    search: []
```

It means user want to remove all search but preserve servers and
options, current nmstate incorrectly treat this as purge also.

This patch only treat these two as purge.

```yml
dns-resolver:
  config: {}
```

and

```yml
dns-resolver:
  config:
    server: []
    search: []
    options: []
```

liangwen12year commented on libnmstate/nm/dns.py:
```diff
@@ -158,7 +158,11 @@ def get_dns_config_iface_names(acs_and_ipv4_profiles, acs_and_ipv6_profiles):
+            ip_profile.props.dns
+            or ip_profile.props.dns_search
+            or ip_profile.props.dns_options
+        ):
```
This seems to me not related to the commit message, maybe worth another commit ?

cathay4t replied:
Moved to new commit.

cathay4t committed:
dns: Fix DNS option `ndots`, `timeout` and `attempts`

The `ndots`, `timeout` and `attempts` DNS options are allowed to
hold a integer value in the format of `<opt_name>:<int>`. Previously,
nmstate is treating them as invalid DNS option. This patch fix it.

Now this YAML is supported:

```yml
dns-resolver:
  config:
    options:
    - rotate
    - ndots:9
```

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
dns: Fix purging DNS config

When user desires:

```yml
---
dns-resolver:
  config:
    search: []
```

It means user want to remove all search but preserve servers and
options, current nmstate incorrectly treat this as purge also.

This patch only treat these two as purge.

```yml
dns-resolver:
  config: {}
```

and

```yml
dns-resolver:
  config:
    server: []
    search: []
    options: []
```

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
packit: Remove EPEL-8 rpm build

The rust compiler in EPEL-8 is outdated which will not pass the rust
compiling, hence removed.

Signed-off-by: Gris Ge <fge@redhat.com>
